src/app/WriteClient.h

```diff
@@ -138,7 +138,69 @@ class WriteClient : public Messaging::ExchangeDelegate
+     *                          This is controlled by the mTimedWriteTimeoutMs field.
+     *
+     * 2. TIMEDREQUEST FLAG: A boolean field in the WriteRequest message itself that indicates whether
+     *                       the write was preceded by a Timed Request action.
+     *                       This is controlled by the mForceTimedRequestFlag field.
+     *
+     * Normal behavior: When you provide a timeout value to the standard constructor, both happen together:
+     *   - A Timed Request action is sent (controlled by mTimedWriteTimeoutMs)
+     *   - The TimedRequest flag in WriteRequest is set to true (mForceTimedRequestFlag = true)
+     *
+     * This test constructor allows you to decouple these for testing edge cases:
+     *   - aTimedRequest = false: No Timed Request action AND TimedRequest flag is false (normal non-timed write)
+     *   - aTimedRequest = true:  No Timed Request action BUT TimedRequest flag is true
+     *                            (This is invalid behavior that should be rejected by the server, used for negative testing)
+     *
+     * @param[in] aTimedRequest If true, sets the TimedRequest flag in WriteRequest to true without actually
+     *                          performing a Timed Request action. This creates a mismatch that should be rejected
+     *                          by a compliant server.
+     */
+    WriteClient(Messaging::ExchangeManager * apExchangeMgr, Callback * apCallback, bool aTimedRequest) :
+        mpExchangeMgr(apExchangeMgr), mExchangeCtx(*this), mpCallback(apCallback), mForceTimedRequestFlag(aTimedRequest)
+    {
+        assertChipStackLockedByCurrentThread();
+    }
+
+    // Tag type to distinguish the test constructor from the normal constructor
+    struct TestOnlyOverrideTimedRequestFlagTag
+    {
+    };
+
+    /**
+     * TestOnly constructor that allows performing a Timed Request action while setting the TimedRequest flag
+     * in the WriteRequest to false (i.e., sending a TimedRequest message first, but then lying about it in the WriteRequest).
+     *
+     * This tests the third edge case: Timed Request action IS performed, but the TimedRequest flag is set to false.
+     *
+     * Test scenarios enabled by WriteClient constructors:
+     * 1. Normal write (both false):              Action = No,  Flag = False  [Standard constructor with no timeout]
+     * 2. Normal timed write (both true):         Action = Yes, Flag = True   [Standard constructor with timeout]
+     * 3. Flag true, no action (invalid):         Action = No,  Flag = True   [Test constructor taking bool aTimedRequest]
+     * 4. Action present, flag false (invalid):   Action = Yes, Flag = False  [THIS constructor]
+     *
+     * @param[in] aTimedWriteTimeoutMs The timeout for the Timed Request action (WILL be sent)
+     * @param[in] aForceTimedRequestFlag If false, the TimedRequest flag in WriteRequest will be false despite the action
```

Comment:
Again, calling this `aTimedRequestFieldValue` would just make it clear that it's the value....